src/ui/StarRating/StarRating.ts

```diff
@@ -0,0 +1,126 @@
+import 'styling/_StarRating';
+import { $$ } from '../../utils/Dom';
+import { Utils } from '../../utils/Utils';
+import { l } from '../../strings/Strings';
+import { Component } from '../Base/Component';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { exportGlobally } from '../../GlobalExports';
+import { IQueryResult } from '../../rest/QueryResult';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions, IFieldOption } from '../Base/ComponentOptions';
+import { Initialization } from '../Base/Initialization';
+
+const DEFAULT_SCALE = 5;
+
+export interface IStarRatingOptions {
+  rating?: IFieldOption;
+  numberOfRatings?: IFieldOption;
+  ratingScale?: number;
+}
+/**
+ * The `StarRating` component renders a 5-star rating widget.
```

Comment:
Looping in @fbeaudoincoveo for documentation review.

I'd say that right now it's not descriptive enough. It should explain that this is normally used in commerce context.

Reply:
Could remove 5-star since it's an option

Reply:
The widget always only displays five stars. Different scales just allows for a wider range of input values but the result is converted to a 0-5 scale.
I agree that it could be more descriptive